ethclient/ethclient_test.go

```diff
@@ -419,10 +420,20 @@ func testHeader(t *testing.T, chain []*types.Block, client *rpc.Client) {
 			if !errors.Is(err, tt.wantErr) {
 				t.Fatalf("HeaderByNumber(%v) error = %q, want %q", tt.block, err, tt.wantErr)
 			}
-			if got != nil && got.Number != nil && got.Number.Sign() == 0 {
-				got.Number = big.NewInt(0) // hack to make DeepEqual work
+
+			gotBytes, err := rlp.EncodeToBytes(got)
+			if err != nil {
+				t.Fatalf("Error serializing received block header.")
+			}
+			wantBytes, _ := rlp.EncodeToBytes(tt.want)
```

Comment:
Why the `err` isn't here?

Reply:
I think I saw some other examples in the code base that ignore the returned error value. I might as well add it for extra safety.

Reply:
then the following `err !=nil` is unnecessary.